oauth2_provider/oauth2_validators.py

```diff
@@ -288,8 +288,10 @@ def validate_scopes(self, client_id, scopes, client, request, *args, **kwargs):
 
     def get_default_scopes(self, client_id, request, *args, **kwargs):
         default_scopes = get_scopes_backend().get_default_scopes(application=request.client, request=request)
-        return default_scopes
-
+        return (
+            request.client.allowed_scopes if request.client.allowed_scopes
+            else default_scopes)
```

Comment:
Is this method supposed to return a `list` of scopes?  Or just a space-separated string?